chia/wallet/cat_wallet/lineage_store.py

```diff
@@ -17,84 +15,61 @@ class CATLineageStore:
     WalletInterestedStore to keep track of puzzle hashes which we are interested in.
     """
 
-    db_connection: aiosqlite.Connection
-    db_wrapper: DBWrapper
+    db_wrapper: DBWrapper2
     table_name: str
 
     @classmethod
-    async def create(cls, db_wrapper: DBWrapper, asset_id: str, in_transaction=False):
+    async def create(cls, db_wrapper: DBWrapper2, asset_id: str):
         self = cls()
         self.table_name = f"lineage_proofs_{asset_id}"
         self.db_wrapper = db_wrapper
-        self.db_connection = self.db_wrapper.db
-        if not in_transaction:
-            await self.db_wrapper.lock.acquire()
-        try:
-            await self.db_connection.execute(
+        async with self.db_wrapper.write_db() as conn:
+            await conn.execute(
                 (f"CREATE TABLE IF NOT EXISTS {self.table_name}(" " coin_id text PRIMARY KEY," " lineage blob)")
             )
-        finally:
-            if not in_transaction:
-                await self.db_connection.commit()
-                self.db_wrapper.lock.release()
```

Comment:
this pattern of manually commit and acquire the db mutex is removed everywhere. `DBWrapper2` handles committing and locking of the DB under the hood.